Feature/exclude import

Zae opened the pull request:
Hi,

I added the exclude flag to the import command, like already exists on the export command. This way keys that are excluded but still present in the export file are ignored.

This way it's possible to do partial imports and also fixes the problem that I had in #105, that deleted my userGroups even though I ignored them because it exports the keys as an empty array.

Zae committed:
Add --exclude option to import to exclude data that exists in the schema file

With usage information in the README

And tests for datatypes filtering in import

Zae commented:
Meh, something went wrong with the commits, let me rebase :)

Zae commented:
This also kinda fixes #50 but the other way around. :)

bvangennep commented:
Looks good, can you add instructions how to use this feature to the readme please

Zae commented:
Just a little update, did not forget about this. Will add some documentation to the PR asap :)

Zae commented:
Hey @bvangennep,

I just remembered this PR :-o

Added the README section and rebased on top of your latest commit on master.

I basically copied the export section and changed the references to import and then tried to maybe make a new section for excluding in both import and export to reduce duplication, but could not really think of a nice way to do that for this small addition.

boboldehampsink commented:
This is looking good. Can you add tests?

Zae commented:
I sure can, are there already tests for the export bit that I can take a look at?

boboldehampsink commented:
https://github.com/nerds-and-company/schematic/blob/master/tests/Services/SchematicTest.php#L345-L362

Zae commented:
Took me a while to get to this again, sorry for that :)

Added tests and rebased into one commit.

bvangennep commented:
Sorry for getting to this so later

Zae commented:
no worries, I took a long time as well :) thanks for the merge and the great package!
